### PR TITLE
Support using attributes aliases in nested query where condition

### PR DESCRIPTION
**Pull request by penghuo**
*Issue #, if available:*

*Description of changes:*
Support using aliases in the where condition for nested query. Which is aligned with https://partiql.org/ specification.

For example,
p is the aliases of the employee_nested's projects attributes.
```
GET /_opendistro/_sql/?
{
  "query":"""
    SELECT e.name FROM employee_nested as e, e.projects as p WHERE p IS NOT NULL
  """
}
```
The result is 
```
    "hits" : [
      {
        "_index" : "employee_nested",
        "_type" : "_doc",
        "_id" : "1",
        "_score" : 0.0,
        "_source" : {
          "name" : "Bob Smith"
        }
      },
      {
        "_index" : "employee_nested",
        "_type" : "_doc",
        "_id" : "3",
        "_score" : 0.0,
        "_source" : {
          "name" : "Jane Smith"
        }
      }
    ]
```
The explained query in DSL is
```
{"from":0,"size":200,"query":{"bool":{"filter":[{"bool":{"must":[{"nested":{"query":{"bool":{"must_not":[{"bool":{"must_not":[{"exists":{"field":"projects","boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}},"path":"projects","ignore_unmapped":false,"score_mode":"none","boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}},"_source":{"includes":["name"],"excludes":[]}}
```

The test data is
```
{"id":3,"name":"Bob Smith","title":null,"projects":[{"name":"AWS Redshift Spectrum querying","started_year":1990,"address":[{"city":"Seattle","state":"WA"},{"city":"Boston","state":"MA"}]},{"name":"AWS Redshift security","started_year":1999,"address":[{"city":"Chicago","state":"IL"}]},{"name":"AWS Aurora security","started_year":2015}],"comments":[{"date":"2018-06-23","message":"I love New york","likes":56},{"date":"2017-10-25","message":"Today is good weather","likes":22}]}
{"id":4,"name":"Susan Smith","title":"Dev Mgr","projects":[],"comments":[{"date":"2018-06-23","message":"comment_2_1","likes":56},{"date":"2017-10-25","message":"comment_2_2","likes":22}]}
{"id":6,"name":"Jane Smith","title":"Software Eng 2","projects":[{"name":"AWS Redshift security","started_year":1998},{"name":"AWS Hello security","started_year":2015,"address":[{"city":"Dallas","state":"TX"}]}],"comments":[{"date":"2018-06-23","message":"comment_3_1","likes":24},{"date":"2017-10-25","message":"comment_3_2","likes":42}]}
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Identifier.java [hunk 1]
```diff
@@ -51,8 +51,13 @@ void rewrite(Scope scope) {
         }
     }
 
+    /**
+     * return the path of the expr name. e.g.
+     * if (expr.name = p.name), return p
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Identifier.java]: maybe better if we clarify the sample SQL, ex. "expecting `p` returned as path in both `WHERE p.name = 'A'` and `WHERE p IS NULL` cases, in which `expr.name` = `p.name` and `p` separately".

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java [hunk 2]
```diff
@@ -358,6 +358,18 @@ public void testNestedSingleOrderBy() throws IOException {
         assertThat(result, containsString("\"nested\":{\"path\":\"projects\""));
     }
 
+    @Test
+    public void testNestedIsNotNull() throws IOException {
+        String explain = explainQuery(
+                "SELECT e.name " +
```
Comment (davidcui-amzn) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java]: Should the "SELECT" portion be inlined with the rest of the query?

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java [hunk 3]
```diff
@@ -358,6 +358,18 @@ public void testNestedSingleOrderBy() throws IOException {
         assertThat(result, containsString("\"nested\":{\"path\":\"projects\""));
     }
 
+    @Test
+    public void testNestedIsNotNull() throws IOException {
```
Comment (dai-chen) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java]: Could we add another IT case for real execution along with this explain testing? And then this one can be renamed to `testNestedIsNotNullExplain`.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java [hunk 4]
```diff
@@ -99,10 +99,7 @@ private SQLMethodInvokeExpr wrapNestedFunction(SQLExpr expr) {
     String pathFromIdentifier(SQLExpr identifier) {
         String field = Util.extendedToString(identifier);
         int lastDot = field.lastIndexOf(".");
-        if (lastDot == -1) {
-            throw new IllegalStateException("pathFromIdentifier() is being invoked on the wrong field [" + field + "]");
-        }
-        return field.substring(0, lastDot);
+        return lastDot == -1 ? field :field.substring(0, lastDot);
```
Comment (davidcui-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java]: Mostly question: It looks like the IllegalStateException will no longer be thrown with this change? If so, will returning just the field be as descriptive/have the same desired effect?

Comment (penghuo) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java]: Done.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Identifier.java (hunk 1 above)
Comment (penghuo) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Identifier.java]: Done.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java (hunk 4 above)
Comment (penghuo) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java]: Yes. It is desired behaviour. for example,
expecting p returned as path in both WHERE p.name = 'A' and WHERE p IS NULL cases in which expr.name = p.name and p separately.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java (hunk 2 above)
Comment (penghuo) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java]: Done.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java (hunk 3 above)
Comment (penghuo) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java]: Add IT. Done.

**Pull request merged**